src/main/scala/com/fulcrumgenomics/umi/DuplexConsensusCaller.scala

```diff
@@ -117,23 +117,55 @@ class DuplexConsensusCaller(override val readNamePrefix: String,
       errorRatePreUmi         = this.errorRatePreUmi,
       errorRatePostUmi        = this.errorRatePostUmi,
       minReads                = 1,
+      maxReads                = maxReads,
       minInputBaseQuality     = this.minInputBaseQuality,
       minConsensusBaseQuality = PhredScore.MinValue,
       producePerBaseTags      = true
     ))
 
   /**
-    * Returns the MI tag minus the trailing suffix that identifies /A vs /B
+    * Returns the MI tag **with** the trailing suffix that identifies /A vs /B
     */
-  override protected[umi] def sourceMoleculeId(rec: SamRecord): String = {
-    rec.get[String](ConsensusTags.MolecularId) match {
-      case None =>
+  private def sourceMoleculeAndStrandId(rec: SamRecord): String = {
+    // Optimization: speed up retrieving this tag by storing it in the transient attributes
+    rec.transientAttrs.getOrElse[String](ConsensusTags.MolecularId, {
```

Comment:
I don't love `getOrElse` usage where the `else` is side-effecting.  I think it's confusing, and also multi-line default values are hard to read.  Is this slower when structured as a match?  if you really must do it this way can we add a `getOrElseUpdate` to the `TransientAttrs`?

Reply:
I'll change it to use `match` statements.